polaris-core/src/test/java/org/apache/polaris/service/storage/aws/AwsCredentialsStorageIntegrationTest.java

```diff
@@ -79,7 +80,7 @@ public void testGetSubscopedCreds() {
                   .returns(roleARN, AssumeRoleRequest::roleArn);
               return ASSUME_ROLE_RESPONSE;
             });
-    String warehouseDir = "s3://bucket/path/to/warehouse";
+    String warehouseDir = scheme + "://bucket/path/to/warehouse";
```

Comment:
Can we test the mixed use cases? For example, we create a catalog with allowed locations like `s3://bucket1/abc/`, while some of tables paths are with `s3a`. Does credential vending work in that case? Also i'm not sure how much this simulation can catch. It'd be nice to have a test against the real s3 storage for the mixed use cases.

Reply:
> Does credential vending work in that case? Also i'm not sure how much this simulation can catch. It'd be nice to have a test against the real s3 storage for the mixed use cases.

@flyrain Yes from my integration test it works, but your question is about a test in the project itself. 
Do we have any means for performing integration tests with AWS services already in the project? Otherwise this is something we should discuss.

Reply:
The regression tests under dir `regtests` are designed for that. It only run against local file system now. S3, Azure, and GCP tests are skipped. @jbonofre and me are working on an AWS account to enable that. Here is the command to run them. You will see s3 ones are skipped
```
./gradlew run
env POLARIS_HOST=localhost ./regtests/run.sh
```

Reply:
thanks @flyrain will check that out

Reply:
@flyrain I have adjusted the test in `AwsCredentialsStorageIntegrationTest:testGetSubscopedCreds` to validate that the policy that we generate does not have any references to S3A.

This should be sufficient proof on this until we get to the test in `regtests`

Reply:
> Can we test the mixed use cases? For example, we create a catalog with allowed locations like `s3://bucket1/abc/`, while some of tables paths are with `s3a`. Does credential vending work in that case? Also i'm not sure how much this simulation can catch. It'd be nice to have a test against the real s3 storage for the mixed use cases.

These tests have been moved to `PolarisS3InterOperabilityTest`, please check